src/DBFTPlugin/ConsensusContext.cs

```diff
@@ -530,5 +551,49 @@ public void Serialize(BinaryWriter writer)
             writer.WriteNullableArray(ChangeViewPayloads);
             writer.WriteNullableArray(LastChangeViewPayloads);
         }
+
+        /// <summary>
+        /// Create a nonce transaction
+        /// </summary>
+        /// <param name="blockIndex">The index of the persisting block</param>
+        /// <param name="pubkey">The public key of the current account</param>
+        /// <param name="settings">Protocal setting</param>
+        /// <returns>Transaction</returns>
+        private Transaction CreateNonceTransaction(uint blockIndex, byte[] pubkey, ProtocolSettings settings)
+        {
+            var (_, nonce) = GetNonce(keyPair.PrivateKey);
+            var tx = new Transaction()
+            {
+                Version = 0,
+                Nonce = nonce,
+                ValidUntilBlock = blockIndex + settings.MaxValidUntilBlockIncrement,
+                Signers = new[]
+               {
+                    new Signer
+                    {
+                        Account = pubkey.ToScriptHash(),
```

Comment:
`pubkey.ToScriptHash()` gets a PublicKeyHash, not the account scriptHash

Reply:
> `pubkey.ToScriptHash()` gets a PublicKeyHash, not the account scriptHash

`Contract.CreateSignatureRedeemScript(PublicKey).ToScriptHash()`?

Reply:
> Contract.CreateSignatureRedeemScript(PublicKey).ToScriptHash()? @nicolegys

Yes

Reply:
@shargon thank you shargon